src/createServer.js

```diff
@@ -1,3 +1,61 @@
-// Write code here
-// Also, you can create additional files in the src folder
-// and import (require) them here
+const http = require('http');
+const { convertToCase } = require('./convertToCase');
+
+const createServer = () => {
+  return http.createServer((req, res) => {
+    const url = new URL(req.url, `http://${req.headers.host}`);
+    const toCase = url.searchParams.get('toCase');
+    const PATH = url.pathname.slice(1);
+    const errors = [];
+
+    if (!toCase) {
+      errors.push({
+        message:
+          `"toCase" query param is required. ` +
+          `Correct request is: "/<TEXT_TO_CONVERT>` +
+          `?toCase=<CASE_NAME>".`,
+      });
+    } else {
+      const supportedCases = ['SNAKE', 'KEBAB', 'CAMEL', 'PASCAL', 'UPPER'];
+
+      if (!supportedCases.includes(toCase.toUpperCase())) {
+        errors.push({
+          message:
+            'This case is not supported. Available cases: ' +
+            'SNAKE, KEBAB, CAMEL, PASCAL, UPPER.',
+        });
+      }
+    }
+
+    if (!PATH) {
+      errors.push({
+        message:
+          'Text to convert is required. Correct request is: ' +
+          '"/<TEXT_TO_CONVERT>?toCase=<CASE_NAME>".',
+      });
+    }
+
+    if (errors.length > 0) {
+      res.writeHead(400, { 'Content-Type': 'application/json' });
+      res.write(JSON.stringify({ errors }));
+
+      return res.end();
+    }
+
+    const text = convertToCase(PATH, toCase.toUpperCase());
```

Comment:
Ensure that the `convertToCase` function is correctly implemented and imported from `./convertToCase`. The function should accept two parameters: the text to convert and the target case name, both of which are correctly passed here. Verify that the function returns an object with `originalCase` and `convertedText` properties as expected.